solidity/random-beacon/test/ReimbursmentPool.test.ts

```diff
@@ -0,0 +1,468 @@
+          ethers.utils.parseEther("8.0")
+        )
+
+        const thirdPartyBalanceAfter = await provider.getBalance(
+          thirdParty.address
+        )
+        const thirdPartyBalanceDiff = thirdPartyBalanceAfter.sub(
+          thirdPartyBalanceBefore
+        )
+        expect(thirdPartyBalanceDiff).to.be.equal(
+          ethers.utils.parseEther("2.0")
+        )
+      })
+
+      context("when receiver is zero address", () => {
+        it("should revert", async () => {
+          await expect(
+            reimbursementPool.connect(owner).withdraw(42, ZERO_ADDRESS)
+          ).to.be.revertedWith("Receiver's address cannot be zero")
+        })
+      })
+
+      context("when withdrawing more than the pool's balance", () => {
+        it("should revert", async () => {
+          await expect(
+            reimbursementPool
+              .connect(owner)
+              .withdraw(ethers.utils.parseEther("42.0"), ZERO_ADDRESS)
+          ).to.be.revertedWith("Insufficient contract balance")
+        })
+      })
+    })
+  })
+
+  describe("refund", () => {
+    beforeEach(async () => {
+      await createSnapshot()
+
+      await thirdParty.sendTransaction({
+        to: reimbursementPool.address,
+        value: ethers.utils.parseEther("10.0"), // Send 10.0 ETH
+      })
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    context("when contract is not authorized", () => {
+      it("should revert", async () => {
+        await expect(
+          reimbursementPool
+            .connect(thirdParty)
+            .refund(ethers.utils.parseEther("2.0"), thirdParty.address)
+        ).to.be.revertedWith("Contract is not authorized for a refund")
+      })
+    })
+
+    context("when contract is authorized", () => {
+      beforeEach(async () => {
+        await createSnapshot()
+
+        await reimbursementPool
+          .connect(owner)
+          .authorize(contractToAuthorize.address)
+      })
+
+      afterEach(async () => {
+        await restoreSnapshot()
+      })
+
+      context("when tx gas price is lower than the max gas price", () => {
+        it("should refund based on tx.gasprice", async () => {
+          const refundeeBalanceBefore = await provider.getBalance(
+            refundee.address
+          )
+
+          const tx = await reimbursementPool
+            .connect(contractToAuthorize)
+            .refund(50000, refundee.address)
+
+          const refundeeBalanceAfter = await provider.getBalance(
+            refundee.address
+          )
+          const refundeeBalanceDiff = refundeeBalanceAfter.sub(
+            refundeeBalanceBefore
+          )
+          expect(refundeeBalanceDiff).to.be.gt(
+            ethers.utils.parseUnits("73000", "gwei")
+          )
+
+          expect(refundeeBalanceDiff).to.be.lt(
+            ethers.utils.parseUnits("146000", "gwei")
+          )
+
+          await expect(tx).not.to.emit(reimbursementPool, "SendingEtherFailed")
```

Comment:
Let's extract this check to a separate `it` test case.

Reply:
[b2781e1](https://github.com/keep-network/keep-core/pull/2826/commits/b2781e16230fd4935ec19337a6b5cd5792a60526)